source/gui/guiHelper.py

```diff
@@ -458,3 +462,55 @@ class SIPABCMeta(wx.siplib.wrappertype, ABCMeta):
 	"""Meta class to be used for wx subclasses with abstract methods."""
 
 	pass
+
+
+class _WxCallOnMainResult:
+	"""Container to hold either the return value or exception raised by a function."""
+
+	__slots__ = ("result", "exception")
+
+
+# TODO: Rewrite to use type parameter lists when upgrading to python 3.12 or later.
+_WxCallOnMain_P = ParamSpec("_WxCallOnMain_P")
+_WxCallOnMain_T = TypeVar("_WxCallOnMain_T")
+
+
+def wxCallOnMain(
+	function: Callable[_WxCallOnMain_P, _WxCallOnMain_T],
+	*args: _WxCallOnMain_P.args,
+	**kwargs: _WxCallOnMain_P.kwargs,
+) -> _WxCallOnMain_T:
+	"""Call a non-thread-safe wx function in a thread-safe way.
```

Comment:
Might be worth emphasising the blocking nature of this earlier.

```suggestion
	"""Call a non-thread-safe wx function in a thread-safe way.
	Blocks current thread.
```